rules.d/60-ccid.rules

```diff
@@ -0,0 +1,14 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+ACTION=="remove", GOTO="ccid_end"
+
+SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ENV{ID_USB_INTERFACES}=="*:0b????:*", ENV{ID_SMARTCARD_READER}="1"
```

Comment:
Hmm, maybe move this into 60-fido-id.rules. It has a *very* similar rule already:

```
# Tag any CCID device (i.e. Smartcard Reader) as security token
SUBSYSTEM=="usb", ATTR{bInterfaceClass}=="0b", TAG+="security-device"
```

Yes, "fido-id" is a bit of a misnomer for this, but I think i's close enough, and has the right nr.

In fact both rules should probably be merged in some way. Maybe the line with TAG+= should be rewritten to match on ID_SMART_CARDREADER instead. Of course, this means the tagged device will be a slightly different one, but that should be OK for its current use (the tag is used by systemd-cryptsetup to install a device monitor for any form of security device, so that PKCS#11 can be reenumerated when a device is plugged in — but given that the tag is not used by PKCS#11 it shouldn't matter which level this is specifically applied to).

hence, maybe use the following to 60-fido-id.rules:

```
SUBSYSTEM=="usb", ENV{DEVTYPE}=="usb_device", ENV{ID_USB_INTERFACES}=="*:0b????:*", ENV{ID_SMARTCARD_READER}="1"
ENV{ID_SMARTCARD_READER}==1,TAG+="security-device"
```

That way, people can drop in further rules that set ID_SMARTCARD_READER and cryptsetup will pick them up properly too

Reply:
Thanks for the detailed review. I updated my commit (patch and commit message) accordingly. Sorry for not grepping further, I should really have found that comment to begin with (it has both "CCID" and "smartcard" in it...).

I have preserved the comment above the `TAG+="security-device"` rule, as I believe it still makes sense and makes more sense there than above the `ENV{ID_SMARTCARD_READER}="1"` rule.